### ISSUE TITLE
Card 3 Starburst

### ISSUE THREAD

**Issue by gskleres** [gskleres/FruityMod-StS#12]
Card 3

Name: Starburst
Rarity: 2-Uncommon
Type: Attack
Basic [2 cost]: Deal 8 damage twice to an enemy, and 8 damage once to all other enemies.
Upgrade [2 cost]: Deal 11 damage twice to an enemy, and 11 damage once to all other enemies.

**Comment by gskleres** [comment]
Works

**Comment by daviscook477** [comment]
Changed name from Arcane Barrage to Starburst

**Comment by gskleres** [comment]
Class file needs updating in the code from "ArcaneBarrage" to "Starburst".

**Comment by gskleres** [comment]
Old card dumped. New card: Uncommon 2 cost, Deal 14(18) damage to an enemy and 8(10) damage to ALL other enemies.

**Comment by daviscook477** [comment]
So the game doesn't really have a good way to make an attack do two damage types. Only one of the damage types will be properly calculated out with strength, vulnerability, etc... Any way we could make the card read "Deal 14 damage to an enemy and then ALL other enemies lose 8 HP." that way it's clear what's happening when only the first bit of damage is affected by modifiers?

**Comment by daviscook477** [comment]
New implementation/text due to technical limitations:

Card 3

Name: Starburst

Rarity: 2-Uncommon
Type: Attack
Basic [2 cost]: Deal 14 damage to an enemy and then ALL other enemies lose 8 HP.
Upgrade [2 cost]: Deal 18 damage to an enemy and then ALL other enemies lose 10 HP.

**Comment by gskleres** [comment]
@daviscook477 can you explain a bit more why this isn't possible? I bet we can creatively come up with a solution that works. For example, maybe it can behind the scenes apply a new unique status to the enemy chosen which means any damage it takes is increased by the same amount it took minus 2, i.e. 8+6(10+8), and then after the damage is dealt remove the status. Can we make status effects that have  no icon or animation?

**Comment by daviscook477** [comment]
So it's not an issue of the selected enemy being hit twice. That's easy to fix. It's that currently there's no good way to calculate modified damage more than once per card. So I wanted the second bit of damage to be HP loss so it's understood that it won't be modified. I'm going to look at damage calculation today in the source code to see if I can babe if

**Comment by daviscook477** [comment]
Did not intend to close earlier.

**Comment by gskleres** [comment]
@daviscook477 If it's not possible to do it this way, would it be possible to do this? "Deal 8(11) damage twice to an enemy, and 8(11) damage once to all other enemies."

**Comment by daviscook477** [comment]
We could do that.

**Comment by gskleres** [comment]
DESIGN UPDATE: Changed to "Deal 8(11) damage twice to an enemy, and 8(11) damage once to all other enemies."

**Comment by gskleres** [comment]
BUG: Other units are still losing HP instead of taking damage, and the first unit is hit only once instead of twice. The action sequence should be:

1. Attack the selected enemy for 8(11).
2. Attack ALL enemies for 8(11).

Is that possible?

**Comment by gskleres** [comment]
Works!
